k8s/test-deployment.py

```
@@ -0,0 +1,437 @@
+
+        # Count resources by type
+        self._count_resources(resources)
+
+        return True
+
+    def _test_service_selectors(self, resources: Dict[str, Any]):
+        """Test that service selectors match deployment labels"""
+        services = {k: v for k, v in resources.items() if v['kind'] == 'Service'}
+        deployments = {k: v for k, v in resources.items() if v['kind'] in ['Deployment', 'StatefulSet']}
+
+        for service_key, service in services.items():
```

Ruff (0.12.2)

Loop control variable `service_key` not used within loop body

Rename unused `service_key` to `_service_key`

(B007)